app/src/openxr/cpp/OpenXRInputSource.cpp

```diff
@@ -821,6 +822,16 @@ void OpenXRInputSource::Update(const XrFrameState& frameState, XrSpace localSpac
         delegate.SetImmersiveBeamTransform(mIndex, hasPosition ? gripTransform : pointerTransform);
         flags |= device::GripSpacePosition;
         delegate.SetBeamTransform(mIndex, vrb::Matrix::Identity());
+#if defined(CHROMIUM) && defined(HVR)
+        // HVR runtime incorrectly return the same values for grip and aim poses. We circumvent that
+        // by inventing a grip pose. The values are chosen to match the controller dimensions. This
+        // is only needed for Chromium because it requires the grip to be based on the aim. As the
+        // runtime returns invalid values for that case, we need to emulate the aim based on grip.
+        if (renderMode == device::RenderMode::Immersive) {
+            static const auto emulatedGripTransform = vrb::Matrix::Translation({0.0, 0.30, 0.0}).Rotation(vrb::Vector(1.0, 0.0, 0.0), -M_PI / 5);
+            delegate.SetTransform(mIndex, emulatedGripTransform);
```

Comment:
We have set the delegate's transform to the pointerTransform, built with the "incorrect" baseSpace (the localSpace instead of the expected gripSpace). So now we fix this by assigning an emulated grip transform, but we do so only in the case we have an active grip location.

Additionally, we have a code a few lines above that sets the pointerTransform as the ImmersiveBeamTransform in the case the grip location is not tracking the position. Is this correct ? 

Shouldn't we just fix the pointerTransform before ? and use the emulated one, when needed (eg. HVR buggy runtime) in the rest of the code ?

Reply:
Do you mean moving this hack outside the `isPoseActive` check?